renovate-config.js

```diff
@@ -1,14 +1,45 @@
 module.exports = {
-    extends: ["config:recommended", "group:allNonMajor"],
-    major: { dependencyDashboardApproval: true },
-    dependencyDashboard: true,
+    globalExtends: ["config:recommended"], // instead of "extends" this resolve immediately as part of global config
     repositories: ['zowe/api-layer'],
     baseBranches: ["v2.x.x","v3.x.x"],
+    dependencyDashboard: true,
+    packageRules: [
+        {
+            //for v.2.x.x branch ignore grouping from extends preset, find all packages which are patches,
+            // slug them and make PR with name "all patch dependencies"
+            "matchBaseBranches": ["v2.x.x"],
+            "groupName": "all patch dependencies",
+            "groupSlug": "all-patch",
+            "matchPackageNames": ["*"],
+            "matchUpdateTypes": ["patch"],
+        },
+        {
+            //for v.2.x.x make dashboard approval to all major and minor dependencies updates
+            "matchBaseBranches": ["v2.x.x"],
+            "matchUpdateTypes": ["major", "minor"],
+            "dependencyDashboardApproval": true,
+        },
+        {
+            //for v.3.x.x branch find all packages which are minor and patches,
+            // slug them and make PR with name "all non-major dependencies"
+            "matchBaseBranches": ["v3.x.x-renovate"],
```

Comment:
I guess here should be v3.x.x branch
```suggestion
            "matchBaseBranches": ["v3.x.x"],
```